src/producer/future_producer.rs

```diff
@@ -153,9 +154,11 @@ impl<C: ClientContext + 'static> ClientContext for FutureProducerContext<C> {
         self.wrapped_context.error(error, reason);
     }
 
```

Comment:
Since `FutureProducerContext` wraps the inner context, you also need to wrap the constant

```rust
const ENABLE_REFRESH_OAUTH_TOKEN_CALLBACK: bool = C::ENABLE_REFRESH_OAUTH_TOKEN_CALLBACK;
```

Otherwise it falls back to the default, `false`

Reply:
the code that I'm using for my tests is using old `consumer` code, I don't really have the ability to test the consumer side of these changes in the current state of this repo, but from what I can tell it should be fine.

Reply:
Good call, thank you!